src/Sunburst.ts

```diff
@@ -327,53 +322,325 @@ export class Sunburst implements IVisual {
         }
     }
 
-    public enumerateObjectInstances(options: EnumerateVisualObjectInstancesOptions): VisualObjectInstanceEnumeration {
-        const instanceEnumeration: VisualObjectInstanceEnumeration = SunburstSettings.enumerateObjectInstances(
-            this.settings || SunburstSettings.getDefault(),
-            options
-        );
-
-        if (options.objectName === Sunburst.LegendPropertyIdentifier.objectName) {
-            const topCategories: SunburstDataPoint[] = this.data.root.children;
-            this.enumerateColors(topCategories, instanceEnumeration);
+    public getFormattingModel(): powerbi.visuals.FormattingModel {
+        const visual_datacard: powerbi.visuals.FormattingCard = {
+            description: "Settings for the layout of the visual",
+            displayName: "Layout Settings",
+            uid: "visual_datacard_uid",
+            groups: []
         }
 
-        return (<VisualObjectInstanceEnumerationObject>instanceEnumeration).instances || [];
+        const group_group: powerbi.visuals.FormattingGroup = this.getGroupGroup();
+        visual_datacard.groups.push(group_group);
+
+        const tooltip_group: powerbi.visuals.FormattingGroup = this.getTooltipGroup();
+        visual_datacard.groups.push(tooltip_group);
+
+        const legend_group: powerbi.visuals.FormattingGroup = this.getLegendGroup();
+        visual_datacard.groups.push(legend_group);
+
+        return { cards: [visual_datacard] };
```

Comment:
consider using 
https://github.com/microsoft/powerbi-visuals-utils-formattingmodel/blob/main/src/FormattingSettingsService.ts#L49C26-L49C26

Reply:
This would likely make the automated slice colorpicker code more complicated: 
<img width="997" alt="image" src="https://github.com/microsoft/powerbi-visuals-sunburst/assets/97496861/88fd88df-a85a-47b6-9f0d-18a48b6368cf">

And would be quite a lengthy refactor, which I believe might be beyond the scope of an accessibility focused PR?

What are the advantages of using the FormattingSettingsService library other than managing uids automatically which can be done with any uid function? I can move the formatting model code to a separate file if the idea is to keep the model specification separate from the visual logic?

Reply:
>What are the advantages of using the FormattingSettingsService library other than managing uids automatically which can be done with any uid function?
1) Model structure might change over time, there is no need to rely on details as the service does it for you
2) The service handles locales automatically (if has corresponding displayNameKeys)
> I can move the formatting model code to a separate file if the idea is to keep the model specification separate from the visual logic?
Yes, separating formatting model into a file is a good idea.

Reply:
[ae5b160](https://github.com/microsoft/powerbi-visuals-sunburst/pull/95/commits/ae5b1607cd996a3127a70f8ebd29835e145cf310)